File: frontend/src/types/recommendations.ts
export type RecommendationTicket = {
  id: string;
  rank: number;
  primary_numbers: number[];
  secondary_numbers: number[];
  statistical_score: number;
  ai_score: number | null;
  final_score: number;
  feature_summary: Record<string, unknown>;
  tags: Record<string, unknown>;
  explanation: string | null;
  primary_hits?: number | null;
  secondary_hits?: number | null;
  prize_level?: string | null;
};

export type RecommendationRun = {
  id: string;
  job_id: string;
  lottery_type: string;
  target_issue: string | null;
  strategy_profile_id: string;
  data_cutoff_issue: string | null;
  data_snapshot_hash: string | null;
  seed: number | null;
  candidate_count: number;
  ai_status: string;
  ai_provider?: string | null;
  ai_model?: string | null;
  status: string;
  evaluation?: {
    draw_issue?: string | null;
    best_rank?: number | null;
    best_primary_hits?: number | null;
    best_secondary_hits?: number | null;
    any_prize?: boolean;
    prize_rule_version?: string | null;
  } | null;
  metrics: Record<string, unknown>;
  created_at: string;
  finished_at: string | null;
  tickets: RecommendationTicket[];
};

export type RecommendationList = {
  items: RecommendationRun[];
  page: number;
  page_size: number;
  total: number;
  total_pages: number;
};

export type BacktestRun = {
  id: string;
  job_id: string;
  lottery_type: string;
  strategy_profile_id: string;
  start_issue: string;
  end_issue: string;
  seed: number | null;
  baseline_trials: number;
  status: string;
  summary: Record<string, unknown>;
  started_at: string | null;
  finished_at: string | null;
  created_at: string;
};

export type BacktestList = {
  items: BacktestRun[];
  page: number;
  page_size: number;
  total: number;
  total_pages: number;
};

export type AIConfig = {
  id: string;
  name: string;
  provider: string;
  base_url: string;
  model: string;
  has_api_key: boolean;
  api_key_masked: string;
  timeout_seconds: number;
  max_tokens: number;
  is_default: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
};

File: backend/app/schemas/recommendations.py
"""Recommendation API schemas."""

from __future__ import annotations

from datetime import datetime
from typing import Any, Literal
from uuid import UUID

from pydantic import BaseModel, Field

LotteryType = Literal["ssq", "dlt"]


class RecommendationCreateRequest(BaseModel):
    lottery_type: LotteryType
    target_issue: str | None = None
    strategy_profile_id: UUID | None = None
    seed: int | None = None
    candidate_count: int | None = Field(default=None, ge=1000, le=200000)
    enable_ai: bool = True


class RecommendationTicketPublic(BaseModel):
    id: UUID
    rank: int
    primary_numbers: list[int]
    secondary_numbers: list[int]
    statistical_score: float
    ai_score: float | None = None
    final_score: float
    feature_summary: dict[str, Any] = Field(default_factory=dict)
    tags: dict[str, Any] = Field(default_factory=dict)
    explanation: str | None = None
    primary_hits: int | None = None
    secondary_hits: int | None = None
    prize_level: str | None = None

    model_config = {"from_attributes": True}


class RecommendationEvaluationSummary(BaseModel):
    draw_issue: str | None = None
    best_rank: int | None = None
    best_primary_hits: int | None = None
    best_secondary_hits: int | None = None
    any_prize: bool = False
    prize_rule_version: str | None = None


class RecommendationRunPublic(BaseModel):
    id: UUID
    job_id: UUID
    lottery_type: str
    target_issue: str | None
    strategy_profile_id: UUID
    data_cutoff_issue: str | None
    data_snapshot_hash: str | None
    seed: int | None
    candidate_count: int
    ai_status: str
    ai_provider: str | None = None
    ai_model: str | None = None
    status: str
    metrics: dict[str, Any] = Field(default_factory=dict)
    evaluation: RecommendationEvaluationSummary | None = None
    created_at: datetime
    finished_at: datetime | None = None
    tickets: list[RecommendationTicketPublic] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class RecommendationRunListData(BaseModel):
    items: list[RecommendationRunPublic]
    page: int
    page_size: int
    total: int
    total_pages: int